Lombiq.HelpfulLibraries.LinqToDb/LinqToDbQueryExecutor.cs

```diff
@@ -0,0 +1,59 @@
+using Dapper;
+using LinqToDB;
+using LinqToDB.Data;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace Lombiq.HelpfulLibraries.LinqToDb
+{
+    public static class LinqToDbQueryExecutor
+    {
+        public static async Task<IEnumerable<TResult>> LinqQueryAsync<TResult>(
+            this ISession session,
+            Func<ITableAccessor, IQueryable> query)
+        {
+            var transaction = await session.DemandAsync();
+            var convertedSql = ConvertSqlToDialect(session, transaction, query);
+
+            return await transaction.Connection.QueryAsync<TResult>(convertedSql, transaction: transaction);
+        }
+
+        private static string ConvertSqlToDialect(
+            ISession session,
+            IDbTransaction transaction,
+            Func<ITableAccessor, IQueryable> query)
+        {
+            // Generate aliases for final projection.
+            LinqToDB.Common.Configuration.Sql.GenerateFinalAliases = true;
+
+            // Instantiating a linq2db connection object as it is required to start building the query. Note that it
+            // won't create an actual connection with the database yet.
+            var dataProvider = DataConnection.GetDataProvider(
+                GetDatabaseProviderName(session.Store.Dialect.Name),
+                transaction.Connection.ConnectionString);
+
+            using var linqToDbConnection = new LinqToDbConnection(

```

Comment:
Ezt már a régebbi kódnál is kérdeztem: ez nem fog egy új SQL connectiont nyitni?

Reply:
Tudtommal nem.

Reply:
Mivel a meglévő transactiont használjuk fel.

Reply:
Meg átadjuk neki a connstringet is, szóval ebben nem vagyok ennyire biztos.

Reply:
Azt azért kell átadni, hogy egy dataproviderrel is gazdagabb legyen, és abból megszűlje a dialectet és a syntaxot. Új connectiont akkor hozna létre, ha az IQueryAble-t  `.ToListAsync()`-el akarnám indítani. De mi csak a saját transaction QueryAsync functionjével indítjuk a hívást.

Legalábbis szerintem. Hogyan lehet 100%-ra menni?

Reply:
Ezekkel próbálkozva, nem változik a connectionök száma:
```sql
exec sp_who2

SELECT DB_NAME(dbid) AS DBName,
COUNT(dbid) AS NumberOfConnections,
loginame
FROM    sys.sysprocesses
where DB_NAME(dbid) = 'Finitive_Finitive'
GROUP BY dbid, loginame
ORDER BY DB_NAME(dbid)
```

Reply:
Ehh, mégis. De ez a része hozza létre az új csatlakozást:
```csharp
var dataProvider = DataConnection.GetDataProvider(
                GetDatabaseProviderName(session.Store.Dialect.Name),
                transaction.Connection.ConnectionString);
```

Reply:
All good, dev-ben is 3 connection van, abból 1 az SSMS. Másik 2(?) a site maga, ezen és dev branchen is így néz ki a site futásakor:
![kép](https://user-images.githubusercontent.com/19938236/118968796-a0c7fe80-b96c-11eb-8c44-e0c30e8d7e17.png)

Reply:
Fasza akkor.